Encountered an error while opening a large PSD file.

**Describe the bug**
Opening PSD files smaller than 50MB works fine, but encountering errors when opening PSD files larger than 50MB.

```python
psd_file = "/home/zhoupc/template-photoshop/data/fo/subheading/14.psd"
psd_file = PSDImage.open(psd_file)
print(len(psd_file._layers))
```

**Environment**
 - Python [3.11]
 - Version [v1.9.31]

bug detail：
Traceback (most recent call last):
  File "/home/zhoupc/template-photoshop/zhoupc-test/test.py", line 12, in <module>
    psd_file = PSDImage.open(psd_file)
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/api/psd_image.py", line 116, in open
    self = cls(PSD.read(f, **kwargs))
               ^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/__init__.py", line 76, in read
    LayerAndMaskInformation.read(fp, encoding, header.version),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 71, in read
    self = cls._read_body(fp, end_pos, encoding, version)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 82, in _read_body
    layer_info = LayerInfo.read(fp, encoding, version)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 149, in read
    self = cls._read_body(fp, encoding, version)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 160, in _read_body
    channel_image_data = ChannelImageData.read(fp, layer_records)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 864, in read
    items.append(ChannelDataList.read(fp, layer.channel_info))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 891, in read
    items.append(ChannelData.read(fp, c.length - 2, **kwargs))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/psd/layer_and_mask.py", line 921, in read
    compression = Compression(read_fmt("H", fp)[0])
                              ^^^^^^^^^^^^^^^^^
  File "/root/anaconda/lib/python3.11/site-packages/psd_tools/utils.py", line 37, in read_fmt
    assert len(data) == fmt_size, "read=%d, expected=%d" % (len(data), fmt_size)
           ^^^^^^^^^^^^^^^^^^^^^
AssertionError: read=0, expected=2

--- a/src/psd_tools/psd/layer_and_mask.py
+++ b/src/psd_tools/psd/layer_and_mask.py
@@ -1076,7 +1076,18 @@ class ChannelDataList(ListElement):
     ) -> T_ChannelDataList:
         items = []
         for c in channel_info:
-            items.append(ChannelData.read(fp, c.length - 2, **kwargs))
+            if c.length < 2:
+                # length < 2 means no channel data in the standard location
+                # (e.g. pixel data is stored in a tagged block). Don't read.
+                # Note: on save, this channel will be written as a minimal
+                # ChannelData (compression header only, length=2), which
+                # differs from the original length=0. This is still a valid
+                # PSD representation per spec; byte-identical round-tripping
+                # is not guaranteed for these channels.
+                logger.debug("  channel %s: length=%d, skipping read", c.id, c.length)
+                items.append(ChannelData())
+            else:
+                items.append(ChannelData.read(fp, c.length - 2, **kwargs))
         return cls(items)  # type: ignore[arg-type]
 
     @property
@@ -1109,7 +1120,10 @@ class ChannelData(BaseElement):
         cls: type[T_ChannelData], fp: BinaryIO, length: int = 0, **kwargs: Any
     ) -> T_ChannelData:
         compression = Compression(read_fmt("H", fp)[0])
-        data = fp.read(length)
+        # length is c.length - 2 (the 2-byte compression header is excluded).
+        # Clamp to 0: a negative value would cause fp.read() to consume until
+        # EOF, corrupting the file pointer for subsequent channels.
+        data = fp.read(max(0, length))
         return cls(compression=compression, data=data)
 
     def write(self, fp: BinaryIO, **kwargs: Any) -> int:

--- a/src/psd_tools/psd/layer_and_mask.py
+++ b/src/psd_tools/psd/layer_and_mask.py
@@ -1076,15 +1076,24 @@ class ChannelDataList(ListElement):
     ) -> T_ChannelDataList:
         items = []
         for c in channel_info:
-            if c.length < 2:
-                # length < 2 means no channel data in the standard location
+            if c.length == 0:
+                # length=0 means no channel data in the standard location
                 # (e.g. pixel data is stored in a tagged block). Don't read.
                 # Note: on save, this channel will be written as a minimal
                 # ChannelData (compression header only, length=2), which
                 # differs from the original length=0. This is still a valid
                 # PSD representation per spec; byte-identical round-tripping
                 # is not guaranteed for these channels.
-                logger.debug("  channel %s: length=%d, skipping read", c.id, c.length)
+                logger.debug("  channel %s: length=0, skipping read", c.id)
+                items.append(ChannelData())
+            elif c.length == 1:
+                # length=1 is malformed (spec requires 0 or >=2). Consume the
+                # stray byte to keep the file pointer aligned, then fall back
+                # to an empty ChannelData so parsing can continue.
+                logger.warning(
+                    "  channel %s: length=1 is invalid, skipping 1 byte", c.id
+                )
+                fp.read(1)
                 items.append(ChannelData())
             else:
                 items.append(ChannelData.read(fp, c.length - 2, **kwargs))
@@ -1121,9 +1130,14 @@ class ChannelData(BaseElement):
     ) -> T_ChannelData:
         compression = Compression(read_fmt("H", fp)[0])
         # length is c.length - 2 (the 2-byte compression header is excluded).
-        # Clamp to 0: a negative value would cause fp.read() to consume until
-        # EOF, corrupting the file pointer for subsequent channels.
-        data = fp.read(max(0, length))
+        # A negative value here indicates an upstream logic error; warn and
+        # clamp to 0 rather than letting fp.read(negative) consume until EOF.
+        if length < 0:
+            logger.warning(
+                "ChannelData.read: negative length %d, clamping to 0", length
+            )
+            length = 0
+        data = fp.read(length)
         return cls(compression=compression, data=data)
 
     def write(self, fp: BinaryIO, **kwargs: Any) -> int:

--- a/tests/psd_tools/psd/test_layer_and_mask.py
+++ b/tests/psd_tools/psd/test_layer_and_mask.py
@@ -288,6 +288,36 @@ def test_channel_data() -> None:
     check_write_read(ChannelData(data=b"\xff" * 8), length=8)
 
 
+def test_channel_data_list_zero_length_channel() -> None:
+    """Zero-length channel must not advance the file pointer (issue #398).
+
+    Channels with length=0 store their pixel data in a tagged block rather
+    than the standard channel image data section.  The parser must skip them
+    without reading any bytes so subsequent channels are parsed correctly.
+    """
+    # Build a stream containing two normal channels (RAW, 2 bytes each)
+    # preceded by one zero-length channel.  The zero-length entry must be
+    # skipped entirely; if even 1 byte is consumed the stream goes out of
+    # sync and the subsequent reads will fail or return wrong data.
+    normal_data = b"\x00\x00"  # Compression=RAW (0x0000), no payload
+    stream = io.BytesIO(normal_data + normal_data)
+
+    channel_info = [
+        ChannelInfo(id=0, length=0),  # zero-length: no bytes in stream
+        ChannelInfo(id=1, length=2),  # normal: 2 bytes (compression only)
+        ChannelInfo(id=2, length=2),  # normal: 2 bytes (compression only)
+    ]
+    result = ChannelDataList.read(stream, channel_info)
+
+    assert len(result) == 3  # type: ignore[arg-type]
+    assert stream.read() == b"", "file pointer should be at end of stream"
+    # The zero-length channel produces a default ChannelData
+    assert result[0].data == b""  # type: ignore[index]
+    # Subsequent channels were parsed without desync
+    assert result[1].compression == Compression.RAW  # type: ignore[index]
+    assert result[2].compression == Compression.RAW  # type: ignore[index]
+
+
 RAW_IMAGE_3x3_8bit = b"\x00\x01\x02\x01\x01\x01\x01\x00\x00"
 RAW_IMAGE_2x2_16bit = b"\x00\x01\x00\x02\x00\x03\x00\x04"
 

